File: backend/src/main/java/at/ac/tuwien/sepm/assignment/individual/service/OwnerService.java
package at.ac.tuwien.sepm.assignment.individual.service;

import at.ac.tuwien.sepm.assignment.individual.dto.OwnerDto;
import at.ac.tuwien.sepm.assignment.individual.entity.Owner;

import java.util.List;

public interface OwnerService {
    /**
     * Get all owners
     * @return returns list of found owners
     * @throws if you pass the wrong arguments it will throw a HttpClientErrorException.NotFound
     */
    List<Owner> getAllOwner();

    /**
     * Inserts owners
     * @params the prename and the surname of o has to be set email can be null
     * @throws if you pass the wrong arguments it will throw a HttpClientErrorException.NotFound
     */
    void insertOwner(OwnerDto o) ;

    /**
     * Updates owner
     * @params the prename and the surname of o has to be set email can be null
     * @throws if you pass the wrong arguments it will throw a HttpClientErrorException.NotFound
     */
    void updateOwner(OwnerDto o) ;

    /**
     * Deletes owner via id
     * @params ownerID has to be valid
     * @throws if you pass the wrong arguments it will throw a HttpClientErrorException.NotFound
     */
    void deleteOwner(Long ownerID) ;

    /**
     * Get 5 closest named existing owners
     * @param name has to be valid
     * @return if you pass the wrong arguments it will throw a HttpClientErrorException.NotFound
     */
    List<Owner> getOwnersAutoComplete(String name);
}

File: frontend/src/app/service/owner.service.ts
import { HttpClient } from '@angular/common/http';
import { Injectable } from '@angular/core';
import { Observable } from 'rxjs';
import { environment } from 'src/environments/environment';
import { Owner } from '../dto/owner';

const baseUri = environment.backendUrl + '/owners';

@Injectable({
  providedIn: 'root'
})
export class OwnerService {

  constructor(
    private http: HttpClient,
  ) { }

  /**
   * Get all owners
   * @return returns list of found owners
   */
  getAll(): Observable<Owner[]> {
    return this.http.get<Owner[]>(baseUri);
  }

  /**
   * Deletes owner via id
   * @params ownerID has to be valid
   */
  deleteOwner(ownerID: number): Observable<object> {
    return this.http.delete(`${baseUri}/${ownerID}`);
  }

  /**
   * Updates owner
   * @params the prename and the surname of o has to be set email can be null
   */
  editOwner(owner: Owner): Observable<object>{
    return this.http.put(baseUri, owner);
  }

  /**
   * Inserts owners
   * @params the prename and the surname of o has to be set email can be null
   */
  insertOwner(owner: Owner): Observable<object>{
    return this.http.post(baseUri, owner);
  }

  /**
   * Get 5 closest named existing owners
   * @param name has to be valid
   */
  getOwnersAutocomplete(name: string): Observable<Owner[]> {
    return this.http.get<Owner[]>(`${baseUri}/autocomplete?name=${name}`);
  }
}
